flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/restart/ExecutionGraphRestarter.java

```diff
@@ -42,4 +48,23 @@ public Object call() throws Exception {
 			}
 		};
 	}
-}
\ No newline at end of file
+
+	public static void scheduleRestartWithDelay(final ExecutionGraph executionGraph, final long delayBetweenRestartAttemptsInMillis,
+												final ScheduledExecutorService executorService) {
+		ScheduledFuture<Object> future = executorService.schedule(
+			new Callable<Object>() {
+				@Override
+				public Object call() throws Exception {
+					executionGraph.restart();
+					return null;
+				}
+			}, delayBetweenRestartAttemptsInMillis, TimeUnit.MILLISECONDS
+		);
+
+		try {
+			future.get();
```

Comment:
This `get()` operation voids the idea of the executor service because it blocks the calling thread.